File: 프로그래머스/lv3/43105. 정수 삼각형/정수 삼각형.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> t) {
    int k = t.size() - 1;
    reverse(t.begin(), t.end());
    for (int i = 0; i < t.size() - 1; i++) {
        for (int j = 0; j < k; j++)
            t[i + 1][j] = max(t[i][j] + t[i + 1][j], t[i][j + 1] + t[i + 1][j]);
        k--;
    }
    return t[t.size() - 1][0];
}

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.java
import java.util.*;

class Solution {
    public int solution(int[][] arr) {
        for (int i = arr[arr.length - 1].length - 1 - 1; i >= 0; i--) {
            for (int j = arr[i].length - 1; j >= 0; j--) {
                arr[i][j] = Math.max(arr[i][j] + arr[i + 1][j], arr[i][j] + arr[i + 1][j + 1]);
            }
        }
        return arr[0][0];
    }
}